UI/src/view/dynamicAssetForm.js

```diff
@@ -0,0 +1,1010 @@
+    DynamicField.call(this, field, isDisabled);
+    var me = this;
+    var className = assetTypeConfiguration.className;
+
+    me.editModeRender = function (fieldValue, sideCode, setValue, getValue) {
+      var value = _.first(fieldValue, function(values) { return values.value ; });
+      var _value = value ? value.value : field.defaultValue ? field.defaultValue : '';
+
+      me.element = $('' +
+        '<div class="form-group">' +
+        '   <label class="control-label">' + field.label + '</label>' +
+        '   <textarea fieldType = "' + field.type + '" '+ me.required() +' name="' + field.publicId + '" class="form-control ' + className + ' ' + me.disabled() + '" >' + _value  + '</textarea>' +
+        '</div>');
+
+
+        if (!isDisabled &&me.hasDefaultValue())
+          me.setSelectedValue(setValue, getValue);
+
+        me.element.find('textarea').on('keyup', function () {
+          me.setSelectedValue(setValue, getValue);
+        });
+
+      return me.element;
+    };
+  };
+
+  var NumericalField = function(assetTypeConfiguration, field, isDisabled){
+    DynamicField.call(this, field, isDisabled);
+    var me = this;
+    var className = assetTypeConfiguration.className;
+
+    me.hasValidValue = function() {
+      return /^\d+$/.test(me.element.find('input').val());
+    };
+
+    me.isValid = function(){
+      return me.isRequired() && me.hasValue() &&  me.hasValidValue() || (!me.isRequired() && (!me.hasValue() ||  me.hasValue() && me.hasValidValue()));
+    };
+
+    me.editModeRender = function (fieldValue, sideCode, setValue, getValue) {
+      var value = _.first(fieldValue, function(values) { return values.value ; });
+      var _value = value ? value.value : field.defaultValue ? field.defaultValue : '';
+
+      var unit = _.isUndefined(field.unit) ? '' :  '<span class="input-group-addon ' + className + '">' + field.unit + '</span>';
+
+      me.element = $('' +
+          '<div class="form-group">' +
+          '   <label class="control-label">' + field.label + '</label>' +
+          '   <input type="text" name="' + field.publicId + '" fieldType = "' + field.type + '" ' + me.required + ' class="form-control" value="' + _value + '"  id="' + className + '" ' + me.disabled() + '>' +
+          unit +
+          '</div>');
+
+      if(!isDisabled && me.hasDefaultValue())
+          me.setSelectedValue(setValue, getValue);
+
+        me.element.find('input').on('keyup', function () {
+          me.setSelectedValue(setValue, getValue);
+        });
+
+      return me.element;
+    };
+  };
+
+  var ReadOnlyFields = function(){
+    DynamicField.call(this);
+    var me = this;
+
+    me.editModeRender = function (fieldValue) {
+      var value = _.first(fieldValue, function(values) { return values.value ; });
+      var _value = value ? value.value : '-';
+      return $('' +
+        '<div class="form-group">' +
+        '   <label class="control-label">' + field.label + '</label>' +
```

Comment:
field is not declared. should it be fieldValue or something else?

Reply:
yes